File: othello_cpp/src/colorprint.cpp
#include "colorprint.hpp"

void print_color(const std::string& text, Color color) {
    // TODO: figure out a nicer way to do this...
    switch (color) {
    case Color::GREY:
        std::cout << termcolor::grey;
        break;
    case Color::RED:
        std::cout << termcolor::red;
        break;
    case Color::GREEN:
        std::cout << termcolor::green;
        break;
    case Color::YELLOW:
        std::cout << termcolor::yellow;
        break;
    case Color::BLUE:
        std::cout << termcolor::blue;
        break;
    case Color::MAGENTA:
        std::cout << termcolor::magenta;
        break;
    case Color::CYAN:
        std::cout << termcolor::cyan;
        break;
    case Color::WHITE:
        std::cout << termcolor::white;
        break;
    }
    std::cout << text << termcolor::reset;
}

void print_bold(const std::string& text, Color color) {
    std::cout << termcolor::bold;
    print_color(text, color);
}

File: othello_python/colorprint.py
"""
Color print helper
Akseli Lukkarila
2019
"""
import colorama
colorama.init()


class Color:
    """Nicer interface to Colorama colors."""
    blue = colorama.Fore.BLUE
    cyan = colorama.Fore.CYAN
    green = colorama.Fore.GREEN
    magenta = colorama.Fore.MAGENTA
    red = colorama.Fore.RED
    white = colorama.Fore.WHITE
    yellow = colorama.Fore.YELLOW


def get_color(text, color=Color.white, bold=False):
    return f"{colorama.Style.BRIGHT if bold else ''}{color}{text}{colorama.Style.RESET_ALL}"


def print_bold(text, color=Color.white):
    print(get_color(text, color, True))


def print_color(text, color=Color.white, bold=False):
    print(get_color(text, color, bold))


def print_error(text):
    print(f"{get_color('Error:', Color.red)} {text}")
